### PR TITLE
tidy issue template update for github changes

### PR DESCRIPTION
**Pull request by Maschette**
This addresses the issue #756. The main changes  were adding a "title" and "about" header to the issue template, and moving the template into a folder called "ISSUE_TEMPLATE", and update the function to account for the directory change.. 

The latter is the important one as if it is not in this folder than for some reason github thinks you are still using an old version. It also gives the location to consider multiple reporting templates as highlighted by @maxheld83 in issue #755

### CODE REVIEW

File: R/tidyverse.R [hunk 1]
```diff
@@ -198,7 +198,7 @@ use_tidy_issue_template <- function() {
   use_directory(".github", ignore = TRUE)
   use_template(
     "tidy-issue.md",
-    ".github/ISSUE_TEMPLATE.md"
+    ".github/ISSUE_TEMPLATE/ISSUE_TEMPLATE.md"
```
Comment (lionel-) [inline R/tidyverse.R]: Maybe the file should be in lower case, like in the [github documentation](https://help.github.com/en/articles/manually-creating-a-single-issue-template-for-your-repository)? Seems a bit less shouty :)

Comment (Maschette) [reply R/tidyverse.R]: I just kept it the same as what is was but I am happy to change it if you want.

Comment (lionel-) [reply R/tidyverse.R]: What do you think @batpigandme?

Comment (batpigandme) [reply R/tidyverse.R]: For sure! Their old rec was to "yell" `ISSUE_TEMPLATE`, so I'm happy to see it lowercase-ified!

**Comment by lionel-** [comment]
Thanks!

**Comment by Maschette** [comment]
Thanks @lionel- and @batpigandme!

File: .github/ISSUE_TEMPLATE/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -1,9 +1,12 @@
+---
```
Comment (jennybc) [inline .github/ISSUE_TEMPLATE/ISSUE_TEMPLATE.md]: This sort of change needs to happen in `inst/templates/tidy-issue.md`. Then we could apply the revised function to usethis itself, to change its issue template(s).

File: R/tidyverse.R [hunk 3]
```diff
@@ -198,7 +198,7 @@ use_tidy_issue_template <- function() {
   use_directory(".github", ignore = TRUE)
```
Comment (jennybc) [inline R/tidyverse.R]: What about the `.github/ISSUE_TEMPLATE` subdirectory?

**Comment by jennybc** [comment]
OK I'm just catching up now and I see that @lionel- made subsequent commits addressing some of these points, i.e. the subdirectory and the tests.

So that just leaves the matter of editing the template used by `use_tidy_issue_template()` vs. the issue template in the usethis repo.

**Comment by Maschette** [comment]
I sent a separate pull request which copied the issue template header from the usethis repo into the issue template used by use_tidy_issue_template(). What I am not sure on as I am not familiar with use_directory is making sure that when you use `use_tidy_issue_template()` for a package it will put the template in .github/ISSUE_TEMPLATE/ which is where it needs to be for github to not complain.

**Pull request merged**